src/clients/c++/perf_client.cc

```diff
@@ -135,14 +135,30 @@ using TimestampVector =
 namespace {
 
 volatile bool early_exit = false;
+// The flag is set when there are no in-flight inferences to complete.
+volatile bool finished_inferencing = false;
 
 void
 SignalHandler(int signum)
 {
-  std::cout << "Interrupt signal (" << signum << ") received." << std::endl
-            << "Waiting for in-flight inferences to complete." << std::endl;
-
-  early_exit = true;
+  // Upon invoking the SignalHandler for the first time early_exit flag is
+  // invoked. On the subsequent invocations, if there are no in-flight inference
+  // requests, the program exits.
+  if (!early_exit) {
+    std::cout << "Interrupt signal (" << signum << ") received." << std::endl
+              << "Initializing early exit..." << std::endl;
```

Comment:
Would "Waiting for in-flight inferences to complete..." be more useful?

Reply:
The client was originally waiting for the inflight inferences before exiting so I decided to carry on the behavior. But it makes more sense to exit immediately when a second interrupt is received. Hence, now perf_client would not wait for all in-flight inferences if second request is raised.

Reply:
On the first Ctrl-C perf_client does/should wait for all in-flight inferences to complete. Can we just keep the original "Waiting for in-flight inferences to complete."?

Reply:
Added the message.